pkg/service/repair/plan.go

```diff
@@ -140,104 +154,111 @@ func (p *plan) Units() []Unit {
 	return out
 }
 
-// SetMaxParallel sets maximal repair parallelism.
-func (p *plan) SetMaxParallel(dcMap map[string][]string) {
-	var max int
-	for _, kp := range p.Keyspaces {
-		// Max parallel is equal to the greatest max keyspace parallel
-		if cand := kp.maxParallel(dcMap); max < cand {
-			max = cand
+func perHostTableSize(ctx context.Context, client *scyllaclient.Client, hkts []hkt) (map[hkt]int64, error) {
+	report, err := client.TableDiskSizeReport(ctx, hkts)
```

Comment:
OMG. I know this is something what was there before, but looks like `client.TableDiskSizeReport` matches reports to `hkts []hkt` basing on the index...
It's blurry...
Why not to change it, to extend the `scyllaclient.HostKeyspaceTable` with size information.
Then, instead of matching by ids, just modify the scyllaclient.HostKeyspaceTable by adding the size, or returning new struct that contains the size ?

I see no reason why this logic is sooo convulted.

This method `perHostTableSize` would not be needed then.